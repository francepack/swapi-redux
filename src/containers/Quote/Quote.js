import React from 'react'
import PropTypes from 'prop-types'

const Quote = (props) => {
  return(
    <div className='quote-box'>
      <div className='crawl'>
        <p className='quote-movie'>{props.title}</p>
        <p className='quote-year'>{props.date}</p>
        <p className='quote'>{props.quote}</p>
      </div>
    </div>  
  )
}

Quote.propTypes = {
  quoteContent: PropTypes.object
}

export default Quote