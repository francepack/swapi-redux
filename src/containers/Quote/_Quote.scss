.quote-box {
  background-color: $smokeBlack;
  border: 4px solid $peach;
  border-radius: 15px;
  box-shadow: 1px 2px 2px 2px;
  color: $Gold;
  display: flex;
  font-size: 260%;
  font-weight: 600;
  height: 200px;
  justify-content: center;
  letter-spacing: 6px;
  line-height: 150%;
  margin: 30px;
  perspective: 400px;
  text-align: justify;
  overflow: hidden;
}

.crawl {
  animation: crawl 120s linear;
  margin: 100px 20px;
  position: relative;
  transform-origin: 50% 100%;
}

.quote {
  font-weight: 700;
}

.quote-movie {
  font-size: 130%;
  font-style: italic;
  margin-bottom: 0;
  text-decoration: underline;
}

.quote-year {
  font-size: 80%;
  margin-top: 0;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}