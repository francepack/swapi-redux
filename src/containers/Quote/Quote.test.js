import React from 'react';
import { enzyme, shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Quote />
    )
  });  
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  });
});