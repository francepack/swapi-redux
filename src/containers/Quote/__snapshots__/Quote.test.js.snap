// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Quote />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="crawl"
      >
        <p
          className="quote-movie"
        />
        <p
          className="quote-year"
        />
        <p
          className="quote"
        />
      </div>,
      "className": "quote-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="quote-movie"
          />,
          <p
            className="quote-year"
          />,
          <p
            className="quote"
          />,
        ],
        "className": "crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "quote-movie",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "quote-year",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "quote",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
        >
          <p
            className="quote-movie"
          />
          <p
            className="quote-year"
          />
          <p
            className="quote"
          />
        </div>,
        "className": "quote-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="quote-movie"
            />,
            <p
              className="quote-year"
            />,
            <p
              className="quote"
            />,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "quote-movie",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "quote-year",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "quote",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
